testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java

```diff
@@ -1175,7 +1175,12 @@ public void testRenameColumn()
     @Test
     public void testInsertIntoNotNullColumn()
     {
-        skipTestUnless(hasBehavior(SUPPORTS_CREATE_TABLE) && hasBehavior(SUPPORTS_INSERT_NOT_NULL_COLUMN));
+        skipTestUnless(hasBehavior(SUPPORTS_CREATE_TABLE));
+
+        if (!hasBehavior(SUPPORTS_NOT_NULL_CONSTRAINT)) {
+            assertQueryFails("CREATE TABLE not_null_constraint (not_null_col INTEGER NOT NULL)", ".* does not support non-null column for column name 'not_null_col'");
```

Comment:
You can add `"\QCatalog '"+  getSession().getCatalog().orElseThrow() +"'` to expected message.